motionplan/motionPlanner.go

```diff
@@ -381,3 +382,81 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
+	checkFrame frame.Frame,
+	plan []map[string][]frame.Input,
+	worldState *frame.WorldState,
+	fs frame.FrameSystem,
+	currentPosition spatialmath.Pose,
+	errorState spatialmath.Pose,
+) error {
+	// ensure that we can actually perform the check
+	if len(plan) < 1 {
```

Comment:
why now only one element when before it was 2?

Reply:
As the robot goes through the plan, we remove waypoints the robot has reached.
Suppose we are almost at the goal, we would then only have one waypoint left to navigate to.
Below, we pre-pend the currentPosition to the plan, thus results in a plan which always has length 2 - however this is downstream. The pre-pending must be downstream of this check since do not know which inputs would correspond to the currentPosition pose.

Reply:
Thanks for explaining! Where do we do that waypoint removal?

Reply:
removing a waypoint once we reach it from the plan is not functionality that exists in main yet. :/